hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/hdds/scm/TestSCMDbCheckpointServlet.java

```diff
@@ -0,0 +1,247 @@
+        .setOmId(omId)
+        .build();
+    cluster.waitForClusterToBeReady();
+    scmMetrics = cluster.getStorageContainerManager().getMetrics();
+  }
+
+  /**
+   * Shutdown MiniDFSCluster.
+   */
+  @After
+  public void shutdown() {
+    if (cluster != null) {
+      cluster.shutdown();
+    }
+  }
+
+  @Test
+  public void testDoGet() throws ServletException, IOException {
+
+    File tempFile = null;
+    try {
+      SCMDBCheckpointServlet scmDbCheckpointServletMock =
+          mock(SCMDBCheckpointServlet.class);
+
+      doCallRealMethod().when(scmDbCheckpointServletMock).init();
+      doCallRealMethod().when(scmDbCheckpointServletMock).initialize(
+          cluster.getStorageContainerManager().getScmMetadataStore().getStore(),
+          cluster.getStorageContainerManager().getMetrics());
+
+      HttpServletRequest requestMock = mock(HttpServletRequest.class);
+      HttpServletResponse responseMock = mock(HttpServletResponse.class);
+
+      ServletContext servletContextMock = mock(ServletContext.class);
+      when(scmDbCheckpointServletMock.getServletContext())
+          .thenReturn(servletContextMock);
+
+      when(servletContextMock.getAttribute(OzoneConsts.SCM_CONTEXT_ATTRIBUTE))
+          .thenReturn(cluster.getStorageContainerManager());
+      when(requestMock.getParameter(OZONE_DB_CHECKPOINT_REQUEST_FLUSH))
+          .thenReturn("true");
+      doNothing().when(responseMock).setContentType("application/x-tgz");
+      doNothing().when(responseMock).setHeader(Matchers.anyString(),
+          Matchers.anyString());
+
+      tempFile = File.createTempFile("testDoGet_" + System
+          .currentTimeMillis(), ".tar.gz");
+
+      FileOutputStream fileOutputStream = new FileOutputStream(tempFile);
+      when(responseMock.getOutputStream()).thenReturn(
+          new ServletOutputStream() {
+            @Override
+            public boolean isReady() {
+              return true;
+            }
+
+            @Override
+            public void setWriteListener(WriteListener writeListener) {
+            }
+
+            @Override
+            public void write(int b) throws IOException {
+              fileOutputStream.write(b);
+            }
+          });
+
+      doCallRealMethod().when(scmDbCheckpointServletMock).doGet(requestMock,
+          responseMock);
+
+      scmDbCheckpointServletMock.init();
+      long initialCheckpointCount = scmMetrics.getNumCheckpoints();
+
+      scmDbCheckpointServletMock.doGet(requestMock, responseMock);
+
+      Assert.assertTrue(tempFile.length() > 0);
+      Assert.assertTrue(
+          scmMetrics.getLastCheckpointCreationTimeTaken() > 0);
+      Assert.assertTrue(
+          scmMetrics.getLastCheckpointStreamingTimeTaken() > 0);
+      Assert.assertTrue(scmMetrics.getNumCheckpoints() >
+          initialCheckpointCount);
+    } finally {
+      FileUtils.deleteQuietly(tempFile);
+    }
+
+  }
+
+  @Test
+  public void testWriteCheckpointToOutputStream() throws Exception {
```

Comment:
This test can be removed. It is already present in TestOMDbCheckpointServlet.

Reply:
yup, will remove.